### PR TITLE
Registration Page for Hacker

### PR DESCRIPTION
**Pull request by uy-seng**
https://www.figma.com/design/BSOPrukZwW5erkftrHb30y/HackPortal?node-id=5417-113&t=FYlGPwdAmGSOLGYa-0

### CODE REVIEW

File: pages/register.tsx [hunk 1]
```diff
@@ -58,9 +60,10 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
     formInitialValues['permissions'] = user?.permissions || ['hacker'];
   }, []);
 
-  useEffect(() => {
-    checkRedirect();
-  }, [user]);
+  // disbale this for testing
+  // useEffect(() => {
+  //   checkRedirect();
+  // }, [user]);
```
Comment (DedsecKnight) [inline pages/register.tsx]: Enable this.

File: pages/register.tsx [hunk 2]
```diff
@@ -125,9 +128,10 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
     );
   }
 
-  if (!user) {
-    router.push('/');
-  }
+  // disable this for testing
+  // if (!user) {
+  //   router.push('/');
+  // }
```
Comment (DedsecKnight) [inline pages/register.tsx]: ^

File: pages/register.tsx [hunk 3]
```diff
@@ -190,19 +194,11 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
         <link rel="icon" href="/favicon.ico" />
       </Head>
       <section className="pt-4 pl-4">
+        {/* TODO: update */}
```
Comment (DedsecKnight) [inline pages/register.tsx]: Will this need to be addressed before merge?

**Review by DedsecKnight — CHANGES_REQUESTED** [review]
It would be great if the textboxes look like the figma as well, but don't worry about it if that takes too much time.

File: pages/register.tsx (hunk 1 above)
Comment (uy-seng) [reply pages/register.tsx]: this is enabled in the latest commit

File: pages/register.tsx (hunk 3 above)
Comment (uy-seng) [reply pages/register.tsx]: forgot to remove this comment, will remove it

**Comment by uy-seng** [comment]
the textbox is the same as the one in the figma design or is there anything i am missing?

**Comment by DedsecKnight** [comment]
> the textbox is the same as the one in the figma design or is there anything i am missing?

Nope, you're right. I just realized that it won't look like the figma until you actually click the textbox. Regardless, feel free to mark this as resolved.

**Review by DedsecKnight — APPROVED** [review]
:100:

**Review by raywa04 — APPROVED** [review]
lgtm!

**Pull request merged**